xbmc/platform/android/activity/XBMCApp.cpp

```diff
@@ -867,6 +890,36 @@ void CXBMCApp::onNewIntent(CJNIIntent intent)
   }
 }
 
+void CXBMCApp::onActivityResult(int requestCode, int resultCode, CJNIIntent resultData)
+{
+  for (auto it = m_activityResultEvents.begin(); it != m_activityResultEvents.end(); ++it)
+  {
+    if ((*it)->GetRequestCode() == requestCode)
+    {
+      m_activityResultEvents.erase(it);
+      (*it)->SetResultCode(resultCode);
+      (*it)->SetResultData(resultData);
+      (*it)->Set();
+      break;
+    }
+  }
+}
+
+int CXBMCApp::WaitForActivityResult(const CJNIIntent &intent, int requestCode, CJNIIntent &result)
+{
+  int ret = 0;
+  CActivityResultEvent* event = new CActivityResultEvent(requestCode);
```

Comment:
Well, still waiting for a reference that it has been "deprecated".
You're obviously welcome to use the wrappers if you like to, or push for a code guideline enforcement ;)

For the record, I'm the one who moved Kodi to c++11 just 2 years ago (https://github.com/xbmc/xbmc/commit/d3c46d246d932849ffb6641427b1278a9d4cd75b) so I'm not a complete dinosaur :)